### PR TITLE
Try new strategy for better rendering of documentation with gopages

### PR DESCRIPTION
**Pull request by aulorbe**
## Problem

Our current documentation is quite ugly (https://sdk.pinecone.io/go/). We generated it by running `godoc -url`, which basically prints standard html to a file. We then pushed this static file to `sdk-docs` via that repo's README instructions. 

## Solution

There is a Go module called `gopages`: https://johnstarich.com/go/gopages/pkg/github.com/johnstarich/go/gopages/ that seemingly renders things better. We should try that to see if it looks nicer.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] Infrastructure change (CI configs, etc)
- [ ] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

## Test Plan

Need to merge to main, then trigger the documentation workflow in the `sdk-docs` repo to see the rendering.

### CODE REVIEW

File: .github/workflows/build-and-publish-docs.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
         env:
           SSH_DEPLOY_KEY: ${{ secrets.SSH_DEPLOY_KEY }}
         with:
-          source-directory: docs
+          source-directory: dist/pkg/github.com/pinecone-io/go-pinecone/pinecone
```
Comment (aulorbe) [inline .github/workflows/build-and-publish-docs.yml:28]: `gopages` automatically creates a `dist` dir (and the nested dirs in this path) when run.

Comment (aulorbe) [reply .github/workflows/build-and-publish-docs.yml:28]: the last `pinecone` dir contains a generated `index.html` file, which is what we'll push to `sdk-docs`.

Comment (austin-denoble) [reply .github/workflows/build-and-publish-docs.yml:28]: Interesting, so all the static site content is within that path after running the utility?

**Review by austin-denoble — APPROVED** [review]
Seems reasonable, let's see how it goes. 🤞

File: .github/workflows/build-and-publish-docs.yml (hunk 1 above)
Comment (aulorbe) [reply .github/workflows/build-and-publish-docs.yml:28]: It looks that way on my local, yes!

**Pull request merged**